source/Calamari.Tests/Fixtures/PackageDownload/PackageDownloadFixture.cs

```diff
@@ -75,16 +74,71 @@ public void TearDown()
             AssertStagePackageOutputVariableSet(result, FeedzPackage);
-            result.AssertOutput("Package {0} v{1} successfully downloaded from feed: '{2}'", FeedzPackage.PackageId, FeedzPackage.Version, PublicFeedUri);
+            result.AssertOutput("Package {0} v{1} successfully downloaded from feed: '{2}'", FeedzPackage.PackageId, FeedzPackage.Version, FeedzUri);
+        }
+
+#if USE_NUGET_V2_LIBS
+        [TestCase("1.0.1", "8950c54441b67dcd8ae54b4dbb0fe3735be85a9b")]
+        [TestCase("1.0.0-alpha02", "9e4e6db295601a70f98f2f0c8bf72c0139eec182")]
+        [TestCase("1.0.0-alpha.2", "991a595d780593acf75b891c8b24edbfaa3505f2")]
+#else
+        [TestCase("1.0.1", "8950c54441b67dcd8ae54b4dbb0fe3735be85a9b")]
+        [TestCase("1.0.2", "3967afe4c54e5f0626677d2f11606d491a453bca")] // actual package is 1.0.02
+        [TestCase("1.0.02", "3967afe4c54e5f0626677d2f11606d491a453bca")]
+        [TestCase("1.0.0-alpha.2", "991a595d780593acf75b891c8b24edbfaa3505f2")]
+        [TestCase("1.0.0-alpha.3", "bd22572257d5ed956b0dd1716e290fba2b4f407c")]
+        [TestCase("1.0.0-alpha.3+metadata", "bd22572257d5ed956b0dd1716e290fba2b4f407c")]
+#endif
+        [RequiresMonoVersion480OrAboveForTls12]
+        [RequiresMinimumMonoVersion(5, 12, 0, Description = "HttpClient 4.3.2 broken on Mono - https://xamarin.github.io/bugzilla-archives/60/60315/bug.html#c7")]
+        public void ShouldDownloadArtifactoryNuGetPackage(string packageVersion, string hash)
+        {
+            SuccessfulPackageDownloadTest("artifactory-nuget", "Artifactory.Test.NuGet", packageVersion, ArtifactoryUri, hash);
+        }
+
+#if USE_NUGET_V2_LIBS
+        [TestCase("1.0.2")] // actual package is 1.0.02
+        [TestCase("1.0.0-alpha2")] // actual package is 1.0.0-alpha02
+        [TestCase("1.0.0-alpha.3")] // actual package is 1.0.0-alpha.3+metadata
+        [TestCase("1.0.0-alpha.3+metadata")] // artifactory-specific nuget issue when using metadata
+#else
+        [TestCase("1.0.0-alpha2")] // actual package is 1.0.0-alpha02
+#endif
+        [RequiresMonoVersion480OrAboveForTls12]
+        [RequiresMinimumMonoVersion(5, 12, 0, Description = "HttpClient 4.3.2 broken on Mono - https://xamarin.github.io/bugzilla-archives/60/60315/bug.html#c7")]
+        public void WillFailToDownloadArtifactoryNuGetPackage(string packageVersion)
+        {
+            UnsuccessfulPackageDownloadTest("artifactory-nuget", "Artifactory.Test.NuGet", packageVersion, ArtifactoryUri);
+        }
+
+        [TestCase("1.0.1", "6ff3395878566eba2b8bef307bd8845be5cb43ff")]
+        [TestCase("1.0.2", "d5ed35df167a8d635f43c890bcba6dd124ef77fb")] // actual package is 1.0.02
+        [TestCase("1.0.02", "d5ed35df167a8d635f43c890bcba6dd124ef77fb")]
+        [TestCase("1.0.0-alpha.2", "eb0ffcc9f614e83d3a69f0963b3da9a6460e75df")]
+        [TestCase("1.0.0-alpha.3", "d148ab7b4caa577b3f4b84960fef13a649182353")]
+        [TestCase("1.0.0-alpha.3+metadata", "d148ab7b4caa577b3f4b84960fef13a649182353")]
```

Comment:
The fact that these package version scenarios work for feedz.io but don't for artifactory even with the V2 libs (presumably running under netframework) suggest to me that we probably need to do a bit more looking into why this is happening rather than just applying a tactical fix of upgrading the NuGet lib for netcore only as we haven't really solved the underlying problem.